src/core/thread/mle_router.cpp

```diff
@@ -2056,64 +2045,67 @@ ThreadError MleRouter::HandleChildIdRequest(const Message &aMessage, const Ip6::
     // Remove from router table
     for (int i = 0; i <= kMaxRouterId; i++)
     {
-        if (mRouters[i].mState != Neighbor::kStateInvalid &&
-            memcmp(&mRouters[i].mMacAddr, &macAddr, sizeof(mRouters[i].mMacAddr)) == 0)
+        if (mRouters[i].GetState() != Neighbor::kStateInvalid &&
+            memcmp(&mRouters[i].GetExtAddress(), &macAddr, sizeof(macAddr)) == 0)
```

Comment:
That operator is not yet defined, was planning to do that general cleanup a separate PR.